packages/ns-api/files/ns.qos

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/python3
+
+#
+# Copyright (C) 2023 Nethesis S.r.l.
+# SPDX-License-Identifier: GPL-2.0-only
+#
+
+import json
+import numbers
+import sys
+
+from euci import EUci
+from nethsec import utils
+from nethsec.utils import ValidationError
+
+
+def validate(data):
+    if 'interface' not in data:
+        raise ValidationError('name', 'required')
+    if 'disabled' not in data:
+        raise ValidationError('disabled', 'required')
+    if data['disabled'] not in [True, False]:
+        raise ValidationError('disabled', 'invalid')
+    if 'upload' not in data:
+        raise ValidationError('upload', 'required')
+    if not isinstance(data['upload'], numbers.Number):
+        raise ValidationError('upload', 'invalid')
+    if 'download' not in data:
+        raise ValidationError('download', 'required')
+    if not isinstance(data['download'], numbers.Number):
+        raise ValidationError('download', 'invalid')
+
+
+cmd = sys.argv[1]
+
+if cmd == 'list':
+    print(json.dumps({
+        'list': {},
+        'add': {
+            'interface': 'String',
+            'disabled': False,
+            'upload': 0,
+            'download': 0,
+        },
+        'delete': {
+            'interface': 'String'
+        },
+        'edit': {
+            'interface': 'String',
+            'disabled': False,
+            'upload': 0,
+            'download': 0,
+        },
+        'set-disabled': {
```

Comment:
What about changing the name of this function to something like `toggle-status`?

Reply:
I've given thought about that, however `toggle-status` it's inferred that no additional data needs to be passed. With `set-status`, it should be hinted by the api call name.

Just a preference, pick you you like :man_shrugging:

Reply:
`set-status` sounds good!